File: 2019-2020/Curs02/Python/05-secunde.py
#!/usr/bin/env python3
"""
   Programarea Calculatoarelor CA, 2019-2020

   Programul transforma un numar de secunde in zile, ore, minute, secunde.
"""


def main():
    SECUNDE_PE_MINUT = 60
    SECUNDE_PE_ORA = 60 * SECUNDE_PE_MINUT
    SECUNDE_PE_ZI  = 24 * SECUNDE_PE_ORA

    print('Introduceti numarul de secunde:')

    # Citim numarul de secunde
    secunde = int(input())

    # Transformam in zile, ore, minute, secunde
    zile = int(secunde / SECUNDE_PE_ZI)
    secunde %= SECUNDE_PE_ZI
    ore = int(secunde / SECUNDE_PE_ORA)
    secunde %= SECUNDE_PE_ORA
    minute = int(secunde / SECUNDE_PE_MINUT)
    secunde %= SECUNDE_PE_MINUT
    print('{} zile {} ore {} minute {} secunde'.format(zile, ore, minute, secunde))

if __name__ == '__main__':
    main()

# rulare: ./05-secunde.py sau python3 05-secunde.py sau python3 ./05-secunde.py
# Pentru prima variantă, este necesar să avem drepturi de execuție asupra fișierului.
#
# Exemple:
#
# input : 225888
# output: 2 zile 14 ore 44 minute 48 secunde
#
# input : 90061
# output: 1 zile 1 ore 1 minute 1 secunde

File: Curs02/C/05-secunde.c
/* Programarea Calculatoarelor CA, 2019-2020

   Programul transforma un numar de secunde in zile, ore, minute, secunde. */

#include <stdio.h>

#define SECUNDE_PE_MIN 60
#define SECUNDE_PE_ORA (60 * SECUNDE_PE_MIN)
#define SECUNDE_PE_ZI  (24 * SECUNDE_PE_ORA)

int main(void)
{
	unsigned int n;
	scanf("%u", &n);

	unsigned int zile = n / SECUNDE_PE_ZI;
	n %= SECUNDE_PE_ZI;

	unsigned int ore = n / SECUNDE_PE_ORA;
	n %= SECUNDE_PE_ORA;

	unsigned int minute = n / SECUNDE_PE_MIN;
	n %= SECUNDE_PE_MIN;

	unsigned int secunde = n;

	printf("%u zile %u ore %u minute %u secunde\n",
		   zile, ore, minute, secunde);

	return 0;
}

/*  compile: gcc 05-secunde.c -o secunde
    run    : ./secunde

    Exemplu:
    input : 225888
    output: 2 zile 14 ore 44 minute 48 secunde

    input: 90061
    outut: 1 zile 1 ore 1 minute 1 secunde  */
